src/components/home/Post.tsx

```diff
@@ -0,0 +1,177 @@
+/** @jsx jsx */
+import React, { useEffect, useRef, useState } from 'react';
+import {jsx, SxStyleProp} from 'theme-ui';
+import {AiOutlineExpandAlt} from 'react-icons/ai';
+import { SeeMoreButton } from './SeeMoreButton';
+
+export interface IPost {
+  name: string;
+  message: string;
+  photo: {
+    url: string;
+    name: string;
+  };
+}
+
+interface PostProps {
+  post: IPost;
+  style?: SxStyleProp;
+}
+
+/**
+ * A component representing a post. This is clickable on both desktop and mobile view
+ * and expands into a full popup.
+ * 
+ * This is memoized because we do not want the component to re-render when the main board changes
+ * since requesting for the photo is pretty expensive.
+ * 
+ * Desktop view:
+ * - Spans a couple columns depending on the window size. 
+ * - The post is a square.
+ * - If there is a photo, then the message component height is constrained.
+ * - Message does not expand, it will go to the popup if clicked.
+ * 
+ * Mobile view:
+ * - Spans full width.
+ * - The image is a square.
+ * - If the `...See More` button is clicked, the message will expand.
+ * 
+ * @returns A post. 
+ */
+export const Post: React.FC<PostProps> = React.memo(({post, style}) => {
+  //TODO: figure out performance with React.memo and without React.memo
+
+  const {name, message, photo} = post;
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const nameRef = useRef<HTMLDivElement>(null);
+  const messageRef = useRef<HTMLDivElement>(null);
+  const [wrapperWidth, setWrapperWidth] = useState<number>(0);
+  const [nameHeight, setNameHeight] = useState<number>(0);
+  const [messageHeight, setMessageHeight] = useState<number>(0);
+
+  const [overflow, setOverflow] = useState<boolean>(false);
+  const [expand, setExpand] = useState<boolean>(false);
```

Comment:
Comments for these constants would be useful for me to understand what exactly they are right from the getgo.